assets/js/components/ClusterDetails/ClusterSettings.jsx

```diff
@@ -34,15 +51,8 @@ export function ClusterSettings() {
         Cluster Settings for{' '}
         <span className="font-bold">{getClusterName(cluster)}</span>
       </PageHeader>
+      {providerWarningBanners[cluster.provider] ?? null}
```

Comment:
You could externalize this outside the `return` statement doing

```js
const warning = providerWarningBanners[cluster.provider];
```

And then using that in the markup, avoiding the `?? null`